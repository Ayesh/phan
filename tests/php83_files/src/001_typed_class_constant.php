<?php
// FIXME TODO support https://wiki.php.net/rfc/typed_class_constants
class C0 {
    // FIXME update php-ast and tolerant-php-parser fallback
    public const int x = 123;
    public const int bad = 'x';
    public const never more = 'x'; // should always warn
}
echo count(C0::x);
echo count(C0::bad);
echo count(C0::more);
