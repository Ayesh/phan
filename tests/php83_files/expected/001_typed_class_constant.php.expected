%s:9 PhanTypeMismatchArgumentInternalReal Argument 1 ($value) is C0::x of type 123 but \count() takes \Countable|\ResourceBundle|\SimpleXMLElement|array (real type \Countable|array)
%s:10 PhanTypeMismatchArgumentInternalReal Argument 1 ($value) is C0::bad of type 'x' but \count() takes \Countable|\ResourceBundle|\SimpleXMLElement|array (real type \Countable|array)
%s:11 PhanTypeMismatchArgumentInternalReal Argument 1 ($value) is C0::more of type 'x' but \count() takes \Countable|\ResourceBundle|\SimpleXMLElement|array (real type \Countable|array)
